rfcs/1440_generic_planner_node.md

```diff
@@ -0,0 +1,482 @@
+  /// of the channel.
+  ///
+  /// Note: this is in anticipation of changing channels to a multi-producer,
+  /// single-consumer model to resolve #1197.
+  receiver_node: String,
+
+  /// The names of nodes that will receive handles to the write-half of the
+  /// channel.
+  sender_nodes: Vec<String>,
+}
+
+/// A dyanimc node definition template.
+struct NodeTemplate {
+  /// Unique name for the node.
+  name: String,
+
+  /// The template for constructing the label associated with the node using
+  /// the extracted paramters.
+  label: LabelTemplate,
+
+  /// The keys of the files from the config map that must be passed to the node
+  /// as part of the initialisation message. This will be used to generate a
+  /// filtered view of the map that will be sent as part of the initialisation
+  /// message.
+  config_files: Vec<String>,
+
+  /// Whether this node should be the target node for all gRPC invocations
+  /// targeted at the specified label. This is only used if the planner node is
+  /// also responsible for invocation routing.
+  ///
+  /// Note: it should be validated that this value is only true for one node
+  /// per label.
+  grpc_invocation_receiver: bool,
+
+  /// Whether this node should be the target node for all HTTP invocations
+  /// targeted at the specified label. This is only used if the planner node is
+  /// also responsible for invocation routing.
+  ///
+  /// Note: it should be validated that this value is only true for one node
+  /// per label.
+  http_invocation_receiver: bool,
+}
+
+/// A static channel definition.
+struct ChannelTemplate {
+  /// Unique name of the channel.
+  name: String,
+
+  /// The template for constructing the label associated with the channel using
+  /// the extracted parameters.
+  label: LabelTemplate,
+
+  /// The name of the node that will receive the handle for the receiver half
+  /// of the channel.
+  ///
+  /// Note: this is in anticipation of changing channels to a multi-producer,
+  /// single-consumer model to resolve #1197.
+  receiver_node: String,
+
+  /// The names of nodes that will receive handles to the write-half of the
+  /// channel.
+  sender_nodes: Vec<String>,
+}
+
+/// A sub-graph template for dynamically creating nodes and channels based on
+/// parameters extract from the label matching process.
+struct Template {
+  /// The template used for matching the label and extracting parameters.
+  label: LabelTemplate,
+
+  /// The template nodes to create.
+  nodes: Vec<NodeTemplate>,
+
+  /// The template channels to create.
+  channels: Vec<ChannelTemplate>,
+}
+
+struct NodeGraph {
+  /// The nodes that will be created.
+  nodes: Vec<Node>,
+
+  /// The channels that will be created.
+  channels: Vec<Channel>,
+
+  /// The per label sub-graph templates used for dynamic node creation.
+  templates: Vec<Template>,
+}
+
+```
+
+### Initialising New Nodes
+
+After creating a new Wasm node the planner node will send an initialisation
```

Comment:
The planner node was mentioned several times and I assumed it was a Wasm node, but reading this paragraph it seems that it is not? Is it a new type of "native" pseudo node? Sorry if I missed it.

Reply:
It is a wasm node, but hopefully a reusable one where the hash will not change.